interface/src/app/sample_annotation/annotation.js

```diff
@@ -0,0 +1,117 @@
+/**
+ * "adage.sampleAnnotation" module.
+ */
+
+angular.module('adage.sampleAnnotation', [
+  'ui.router',
+  'ui.bootstrap',
+  'greenelab.utils'
+])
+
+.config(['$stateProvider', function($stateProvider) {
+  $stateProvider.state('sampleAnnotation', {
+    url: '/sample_annotation?node&samples',
+    views: {
+      main: {
+        templateUrl: 'sample_annotation/annotation.tpl.html',
+        controller: 'SampleAnnotationCtrl as ctrl'
+      }
+    },
+    data: {pageTitle: 'Sample Annotations'}
+  });
+}])
+
+.controller('SampleAnnotationCtrl', [
+  '$stateParams', '$q', '$http', '$log', 'ApiBasePath', 'errGen',
+  function($stateParams, $q, $http, $log, ApiBasePath, errGen) {
+    var self = this;
+    self.queryStatus = 'Connecting to the server ...';
+
+    // Ensure that the URL includes sample ID(s).
+    if (!$stateParams.samples) {
+      self.queryStatus = 'Please specify sample ID(s) in the URL.';
+      return;
+    }
+
+    var samplesInUrl = $stateParams.samples;
+    self.uniqueAnnotationTypes = [];
+    // Note: Ideally self.uniqueAnnotationTypes should be a hashtable (such as
+    // an object of Set) instead of Array, but Set is only available since ES6.
+
+    self.hasNode = false;
+    var sampleDict = {};
+
+    // Function that sets valid sample IDs asynchronously:
+    var getSampleIDPromise = function() {
+      var deferred = $q.defer();
```

Comment:
As we discussed in Rene's last PR, I can definitely see the advantages of abstracting the $http into a separate service; But on the other hand, I am hesitant about using it everywhere possible. I personally feel that having a service as simple as:
```
myMod.service('foobar', ['$resource', function($resource) {
  return $resource(url);
}]);
```
is an overkill. Instead, in this program, I tried to convert the whole function `getSampleIDPromise` into a service, but because it involves the error handling, I eventually gave up. 

What do you think @rzelayafavila and @cgreene?